lib/cisco_node_utils/bgp_af.rb

```diff
@@ -547,5 +547,183 @@ def redistribute=(should)
     def default_redistribute
       config_get_default('bgp_af', 'redistribute')
     end
+
+    # route target both auto(Getter/Setter/Default)
+    def route_target_both_auto
+      config_get('bgp_af', 'route_target_both_auto', @get_args)
+    end
+
+    def route_target_both_auto=(enable)
+      @set_args[:state] = (enable ? '' : 'no')
+      config_set('bgp_af', 'route_target_both_auto', @set_args)
+      set_args_keys_default
+    end
+
+    def default_route_target_both_auto
+      config_get_default('bgp_af', 'route_target_both_auto')
+    end
+
+    # route target both auto evpn(Getter/Setter/Default)
+    def route_target_both_auto_evpn
+      config_get('bgp_af', 'route_target_both_auto_evpn', @get_args)
+    end
+
+    def route_target_both_auto_evpn=(enable)
+      @set_args[:state] = (enable ? '' : 'no')
+      config_set('bgp_af', 'route_target_both_auto_evpn', @set_args)
+      set_args_keys_default
+    end
+
+    def default_route_target_both_auto_evpn
+      config_get_default('bgp_af', 'route_target_both_auto_evpn')
+    end
+
+    # route target export
+    def route_target_export
+      cmds = config_get('bgp_af', 'route_target_export', @get_args)
+      # config_get returns an array, so cmds == ['1:1','2:2']
+
+      # normalize to a string
+      cmds.empty? ? default_route_target_export : cmds.join(' ')
+    end
+
+    def route_target_export=(should)
+      # 'should'comes in as a string, normalize into array
+      should = should.nil? ? [] : should.split(' ')
```

Comment:
`      should = should.empty? ? [] : should.split(' ')`

same change for the `is.nil?` below